iceoryx_examples/waitset/ice_waitset_sync.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/popo/modern_api/typed_subscriber.hpp"
+#include "iceoryx_posh/popo/user_trigger.hpp"
+#include "iceoryx_posh/popo/wait_set.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "topic_data.hpp"
+
+#include <chrono>
+#include <csignal>
+#include <iostream>
+
+iox::popo::UserTrigger shutdownTrigger;
+
+static void sigHandler(int f_sig [[gnu::unused]])
+{
+    shutdownTrigger.trigger();
+}
+
+class SomeClass
+{
+  public:
+    static void cyclicRun(iox::popo::UserTrigger* trigger)
+    {
+        std::cout << "activation callback\n";
+
+        // after every call we have to reset the trigger otherwise the waitset
+        // would immediately call us again since we still signal to the waitset that
+        // we have been triggered (waitset is state based)
+        trigger->resetTrigger();
+    }
+};
+
+int main()
+{
+    signal(SIGINT, sigHandler);
+    iox::runtime::PoshRuntime::initRuntime("/iox-ex-waitset-sync");
+    std::atomic_bool keepRunning{true};
+
+    iox::popo::WaitSet waitset;
+
+    // attach shutdownTrigger to handle CTRL+C
+    shutdownTrigger.attachTo(waitset);
+
+    // create and attach the cyclicTrigger with a callback to
+    // SomeClass::myCyclicRun
+    iox::popo::UserTrigger cyclicTrigger;
+    cyclicTrigger.attachTo(waitset, 0, SomeClass::cyclicRun);
```

Comment:
What is the 0 for? An unused id? Having to specify this one if we only want to register a static (or free) function is a bit unfortunate I think.

Reply:
I can also create an overload for `attachTo` with only two parameters so that the user does not have to provide some arbitrary id. But I would move this to the next PR.

I noted it in the issue: https://github.com/eclipse/iceoryx/issues/341